discord_slash/model.py

```diff
@@ -194,39 +194,59 @@ def __init__(self, *, state, channel, data, _http: http.SlashCommandRequest, int
+        """
+        _resp = {}
+
+        content = fields.get("content")
+        if content:
+            _resp["content"] = str(content)
+
+        embed = fields.get("embed")
+        embeds = fields.get("embeds")
+        file = fields.get("file")
+        files = fields.get("files")
+
+        if embed and embeds:
+            raise error.IncorrectFormat("You can't use both `embed` and `embeds`!")
+        if file and files:
+            raise error.IncorrectFormat("You can't use both `file` and `files`!")
+        if file:
+            files = [file]
+        if embed:
+            embeds = [embed]
+        if embeds:
+            if not isinstance(embeds, list):
+                raise error.IncorrectFormat("Provide a list of embeds.")
+            elif len(embeds) > 10:
+                raise error.IncorrectFormat("Do not provide more than 10 embeds.")
+            _resp["embeds"] = [x.to_dict() for x in embeds]
+
+        allowed_mentions = fields.get("allowed_mentions")
+        _resp["allowed_mentions"] = allowed_mentions.to_dict() if allowed_mentions else \
+            self._state.allowed_mentions.to_dict() if self._state.allowed_mentions else {}
+
+        await self._http.edit(_resp, self.__interaction_token, self.id, files = files)
+
+        delete_after = fields.get("delete_after")
+        if delete_after:
+            await self.delete(delay=delete_after)
+        if files:
+            for file in files:
+                file.close()
+
+
     async def edit(self, **fields):
         """Refer :meth:`discord.Message.edit`."""
-        try:
-            await super().edit(**fields)
-        except discord.Forbidden:
-            _resp = {}
-
-            content = str(fields.get("content"))
-            if content:
-                _resp["content"] = str(content)
-
-            embed = fields.get("embed")
-            embeds = fields.get("embeds")
-            if embed and embeds:
-                raise error.IncorrectFormat("You can't use both `embed` and `embeds`!")
-            if embed:
-                embeds = [embed]
-            if embeds:
-                if not isinstance(embeds, list):
-                    raise error.IncorrectFormat("Provide a list of embeds.")
-                elif len(embeds) > 10:
-                    raise error.IncorrectFormat("Do not provide more than 10 embeds.")
-                _resp["embeds"] = [x.to_dict() for x in embeds]
-
-            allowed_mentions = fields.get("allowed_mentions")
-            _resp["allowed_mentions"] = allowed_mentions.to_dict() if allowed_mentions else \
-                self._state.allowed_mentions.to_dict() if self._state.allowed_mentions else {}
-
-            await self._http.edit(_resp, self.__interaction_token, self.id)
-
-            delete_after = fields.get("delete_after")
-            if delete_after:
-                await self.delete(delay=delete_after)
+        if "file" in fields or "files" in fields:
+            await self._slash_edit(**fields)
```

Comment:
Everything should be wrapped to try except block since this is for trying discord.py's original function, then if we get 403, which means only slash commands are added, it will use slash command webhook.

Reply:
d.py doesn't support files while editing messages yet, so i'm just skipping it because it'll error every time